boards/common/nxp_s32dbg.board.cmake

```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+board_set_flasher_ifnset(nxp_s32dbg)
```

Comment:
The Python code says: return RunnerCaps(commands={'debug', 'debugserver', 'attach'},, no flash capabilities. 
Isn't setting the flasher to nxp_s32dbg strange? Or is it because a flasher must be set to the s32 dbg runner can tell west the command 'flash' is not supported?

Reply:
that's right, the error the user get is `FATAL ERROR: runner nxp_s32dbg does not support command flash`. This is to avoid having two default runners, one for debug and another for flash command. Instructions are added in board's doc on how to use all the supported runners on this board.